src/NHibernate/IFutureValueAsync.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading.Tasks;
+
+namespace NHibernate
+{
+	public interface IFutureValueAsync<T>
+	{
+		Task<T> GetValue();
```

Comment:
I agree with you. As you already suggested, we should remove the cancellation token parameter from `FutureAsync` and `FutureValueAsync` methods and put it here. [Here](https://github.com/OnurGumus/nhibernate-core/commit/de83d45d2fc2099997ae60b2046dc83c7cc007c1) we can see the changes that we should apply.